src/core/server/opensearch/opensearch_config.ts

```diff
@@ -131,6 +131,7 @@ export const configSchema = schema.object({
   healthCheck: schema.object({ delay: schema.duration({ defaultValue: 2500 }) }),
   ignoreVersionMismatch: schema.boolean({ defaultValue: false }),
   disablePrototypePoisoningProtection: schema.maybe(schema.boolean({ defaultValue: false })),
+  compression: schema.maybe(schema.boolean({ defaultValue: false })),
```

Comment:
I am thinking why don't we make it a string? Though `gzip` is the most common method to compress request.

Reply:
this is actually due to the OpenSearch JS client:
```
https://github.com/opensearch-project/opensearch-js/blob/main/lib/Transport.d.ts#L73
```

Reply:
updated!